test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Builder/LinkGenerationHelpersTest.cs

```diff
@@ -54,6 +54,24 @@ public void GenerateNavigationLink_WorksToGenerateExpectedNavigationLink_ForEnti
             Assert.Equal(expectedNavigationLink, uri.AbsoluteUri);
         }
 
+        [Theory]
+        [InlineData(false, "http://localhost/MyOrders(42)/OrderLines")]
+        [InlineData(true, "http://localhost/MyOrders(42)/NS.MyOrder/OrderLines")]
```

Comment:
Could you also add a test where the last path segment is a type cast?

Reply:
Actually in this test the last segment has a typecast